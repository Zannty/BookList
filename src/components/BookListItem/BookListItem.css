.book-list-item {
  display: flex;
  font-size: 2em;
}
.property-item {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-bottom: 10px;
  margin-left: 1em;
  font-family: "Oswald", sans-serif;
}
.cover {
  flex-shrink: 0;
  width: 240px;
  margin-right: 30px;
}

img {
  max-width: 100%;
}
