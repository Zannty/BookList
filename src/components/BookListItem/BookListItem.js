import React from "react";
import "./BookListItem.css";
const BookListItem = props => {
  const { photo, genre, author, name } = props.book;
  return (
    <div className="book-list-item">
      <div className="cover">
        <img src={photo} />
      </div>
      <div className="property-item">
        <div>Название книги: {name}</div>
        <div>Автор: {author}</div>
        <div>Жанр: {genre}</div>
      </div>
    </div>
  );
};

export default BookListItem;
