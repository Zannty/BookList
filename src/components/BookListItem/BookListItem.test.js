import BookListItem from "./BookListItem";
import React from "react";
describe("BookListItem", () => {
  const props = { book: { photo: "x", genre: "y", author: "z", name: "q" } };
  const container = shallow(<BookListItem {...props} />);

  it("render props", () => {
    expect(container.debug()).toMatchSnapshot();
  });
});
