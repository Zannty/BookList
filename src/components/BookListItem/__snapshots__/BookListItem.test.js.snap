// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListItem render props 1`] = `
"<div className=\\"book-list-item\\">
  <div className=\\"cover\\">
    <img src=\\"x\\" />
  </div>
  <div className=\\"property-item\\">
    <div>
      Название книги: 
      q
    </div>
    <div>
      Автор: 
      z
    </div>
    <div>
      Жанр: 
      y
    </div>
  </div>
</div>"
`;
